lib/mssql.js

```diff
@@ -0,0 +1,215 @@
+import Ajv from "ajv";
+import JSONStream from "JSONStream";
+import {json} from "micro";
+import mssql from "mssql";
+import {notFound, failedCheck, badRequest} from "./errors.js";
+
+const TYPES = mssql.TYPES;
+
+const pools = new Map();
+
+// TODO: Confirm these are the right SQL verbs.
+const READ_ONLY = new Set(["SELECT", "SHOW DATABASES", "SHOW VIEW", "USAGE"]);
+
+const ajv = new Ajv();
+const validate = ajv.compile({
+  type: "object",
+  additionalProperties: false,
+  required: ["sql"],
+  properties: {
+    sql: {type: "string", minLength: 1},
+    params: {type: "array"},
+  },
+});
+
+// Since we want to work with more than one DB,
+// we will need to use a custom version of the MSSQL pool.
+// See: https://tediousjs.github.io/node-mssql/#connection-pools
+export const mssqlPool = {
+  get: (name, config) => {
+    if (!pools.has(name)) {
+      if (!config) {
+        throw new Error("Pool does not exist");
+      }
+
+      const pool = new mssql.ConnectionPool(config);
+      const close = pool.close.bind(pool);
+      pool.close = (...args) => {
+        pools.delete(name);
+        return close(...args);
+      };
+
+      pools.set(name, pool.connect());
+    }
+
+    return pools.get(name);
+  },
+
+  closeAll: () =>
+    Promise.all(
+      Array.from(pools.values()).map((connect) => {
+        return connect.then((pool) => pool.close());
+      })
+    ),
+};
+
+export function sanitizeForCellTag(string) {
```

Comment:
This won't work in general and we should fix the DatabaseClient to use @ instead of ? so it is not needed.